internal/validator/validator.go

```diff
@@ -15,10 +17,57 @@ type OutpointData struct {
 	Satoshis     int64
 }
 
-type Validator interface {
-	// ValidateEFTransaction Please note that bt.Tx should have all the fields of each input populated.
-	ValidateEFTransaction(tx *bt.Tx, skipFeeValidation bool, skipScriptValidation bool) error
-	ValidateBeef(beef *beef.BEEF, skipFeeValidation bool, skipScriptValidation bool) (*bt.Tx, error)
+type FeeValidation byte
+
+const (
+	NoneFeeValidation FeeValidation = iota
+	StandardFeeValidation
+)
+
+type ScriptValidation byte
+
+const (
+	NoneScriptValidation ScriptValidation = iota
+	StandardScriptValidation
+)
+
+type DefaultValidator interface {
+	ValidateTransaction(ctx context.Context, tx *bt.Tx, feeValidation FeeValidation, scriptValidation ScriptValidation) error
 	IsExtended(tx *bt.Tx) bool
-	IsBeef(txHex []byte) bool
+}
+
+type BeefValidator interface {
+	ValidateTransaction(ctx context.Context, beef *beef.BEEF, feeValidation FeeValidation, scriptValidation ScriptValidation) (*bt.Tx, error)
+}
```

Comment:
Why not make it one interface and use a strategy design pattern here?

Reply:
Let me break the answer into a few points:

1. This is a big PR with a significant number of changes (specifically moving the code around), and I wanted to make it as reviewable as possible. Simply dividing the existing interface into two was a way to limit changes in the client code.
2. I considered creating a unified validation interface, but this would lead to the creation of another input structure and the need to validate this input sturcture inside our specialized validators, which feels like "validation inception" – something I don't like. That said, I believe we can unify them on the client side using the `Adapter` pattern.

Reply:
Makes sense, approved ✅